source/localizable/tutorial/location-service.md

```diff
@@ -0,0 +1,301 @@
+In Super Rentals, we want the ability to provide directions from where the renter is located to his or her various rental options.
+To accomplish this, we need to make use of the browser's native geolocation API to determine our current location,
+as well as interface with google's map and geocoding API to display directions.
+
+To implement this feature, we are going to take advantage of several Ember concepts:
+
+1. A component to display a google map with directions.
+2. Utility functions to interface with Google APIs.
+3. A service to keep track of the browser's current location.
+
+### Making Google Maps Available in Ember
+
+There are several ways to include 3rd party libraries in Ember. See the guides section on [managing dependencies](../../addons-and-dependencies/managing-dependencies/) as a starting point when you need to add one.
+
+Since Google provides its map API as a remote script, we'll use curl to download it into our project's vendor directory.
+
+```shell
+curl -o vendor/gmaps.js https://maps.googleapis.com/maps/api/js?v=3.22
+```
+Once in the vendor directory, the script can be built into our app.
+We'll just need to import it in our build file.
+
+```ember-cli-build.js
+/*jshint node:true*/
+/* global require, module */
+var EmberApp = require('ember-cli/lib/broccoli/ember-app');
+
+module.exports = function(defaults) {
+  var app = new EmberApp(defaults, {
+    // Add options here
+  });
+
+  // Use `app.import` to add additional libraries to the generated
+  // output files.
+  //
+  // If you need to use different assets in different
+  // environments, specify an object as the first parameter. That
+  // object's keys should be the environment name and the values
+  // should be the asset to use in that environment.
+  //
+  // If the library that you are including contains AMD or ES6
+  // modules that you would like to import into your application
+  // please specify an object with the list of modules as keys
+  // along with the exports of each module as its value.
+  app.import('vendor/gmaps.js');
+
+  return app.toTree();
+};
+```
+
+### Loading Maps into Application Code
+
+Now that the google map API library is built into our app, lets create a service to wrap them.
+Services are used to package common concerns and keep shared state across the application
+
+Accessing our maps API through a service will give us several benefits
+
+* It is injected with a service locator pattern,
+  meaning it will abstract the maps API from the code that uses it, allowing for easier refacoring and maintenance
+* It is lazy-loaded, meaning it won't be initialized until its called the first time.
+* It is a singleton, which could allow for potential caching of map data.
```

Comment:
While it's true it is a singleton, I think we need to either explain why that's important or figure out how else to say this in more "beginner-friendly" terminology